packages/upgrade/src/upgrades.js

```diff
@@ -312,4 +312,44 @@ export const upgrades = [
       },
     ],
   },
+  {
+    name: 'featureflag-deprecate-flags-prop',
+    description: `
+    Updates the <FeatureFlags> component usage:
+    1. Deprecates the 'flags' object prop
+    2. Replaces it with individual boolean props for each feature flag
+    3. Removes usage of no longer needed flags (e.g., 'enable-v11-release')
+    
+    Example transformation:
+    Before: <FeatureFlags flags={{ 'flag1': true, 'flag2': false, 'enable-v11-release': true }}>
+    After:  <FeatureFlags flag1 flag2={false}>
+  `,
+
+    migrate: async (options) => {
+      const transform = path.join(
+        TRANSFORM_DIR,
+        'featureflag-deprecate-flags-prop.js'
+      );
+      const paths =
+        Array.isArray(options.paths) && options.paths.length > 0
+          ? options.paths
+          : await glob(['**/*.js', '**/*.jsx'], {
+              cwd: options.workspaceDir,
+              ignore: [
+                '**/es/**',
+                '**/lib/**',
+                '**/umd/**',
+                '**/node_modules/**',
+                '**/storybook-static/**',
+              ],
+            });
+
+      await run({
+        dry: !options.write,
+        transform,
+        paths,
+        verbose: options.verbose,
+      });
+    },
+  },
```

Comment:
This should be moved up into the `migrations` array after `update-carbon-icons-react-import-to-carbon-react`